pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1119,21 +1119,28 @@ func maybeRetryBoot(ctx *domainContext, status *types.DomainStatus) {
 	}
 	defer file.Close()
 
-	if err := hyper.Task(status).VirtualTPMSetup(status.DomainName, agentName, ctx.ps, warningTime, errorTime); err != nil {
-		log.Errorf("Failed to setup virtual TPM for %s: %s", status.DomainName, err)
-		status.VirtualTPM = false
-	} else {
+	err = hyper.Task(status).VirtualTPMSetup(status.DomainName, agentName, ctx.ps, warningTime, errorTime)
+	if err == nil {
 		status.VirtualTPM = true
+		defer func(status *types.DomainStatus) {
+			if status.BootFailed || status.HasError() {
```

Comment:
We have discussed these conditions with @shjala, and it looks like a more reliable way to check that we do not need to terminate the vTMP is `status.Activated` set to `true`, as it guarantees that `ActivateTails` finished successfully.